hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/AttributeSearchService.kt

```diff
@@ -73,6 +78,15 @@ class AttributeSearchService {
     return allTextMatchers
   }
 
+  private fun getAllBooleanMatchers(matcher: List<Matcher>): List<BooleanMatcher> {
+    val allBooleanMatchers = mutableListOf<BooleanMatcher>()
+    matcher.forEach {
+      it.booleanMatchers?.let { booleanMatchers -> allBooleanMatchers.addAll(booleanMatchers) }
+      it.children?.let { children -> allBooleanMatchers.addAll(getAllBooleanMatchers(children)) }
+    }
+    return allBooleanMatchers
+  }
```

Comment:
I might try and make this generic because they're all the same. Something like `private fun getAllMatchers<T : TypeMatcher>(matcher: List<Matcher>): List<T " TypeMatcher>`